### ISSUE TITLE
Switch to a new HTML Renderer

### ISSUE THREAD

**Issue by tresf** [qzind/tray#558]
We're investigating dropping JavaFX from our source code due to the poor support it has for printing (e.g. #532, #268) as well as many reliability issues in regards to printing (#96, #280, #231, #506, #497, #547).

Using an open source renderer:
* Very few exist that use modern standards.
* Ones that do often require us to build them ourselves.

Using a commercial renderer:
* May make the build environment hard to support as open-source

Ideas are welcome.  So far I've found the following projects:
* http://www.webrenderer.com/ (commercial)
* https://bitbucket.org/chromiumembedded/java-cef/src (open source)
* https://www.teamdev.com/jxbrowser (commercial) <br>🚫 prohibits API and library in an open source project.

**Comment by tresf** [comment]
Another idea... what if we leveraged jsPDF under the covers to render HTML?  Then we'd use the reliable PDFBOX for printing and let the browser do the rest?  Note, this idea is slightly stolen from @lukas-w from another code repository. :)

**Comment by tresf** [comment]
@bberenz researched jsPDF and although it's often the first result in an online search, they're [discontinuing the `fromHTML` function](https://github.com/MrRio/jsPDF/issues/347#issuecomment-419229280).

I'm looking into other services and most are commercial.  So far the best I've found quality-wise is https://pdfcrowd.com/.  Without a subscription it'll watermark the content, but the render quality is the best I've found so far.  There's also obvious privacy concerns if we were to farm out any HTML rendering to a 3rd-party provider.
